### PR TITLE
New package: budgie-desktop-10.2.7

### PR DESCRIPTION
**Pull request by mar1n3r0**
The Budgie Desktop is the flagship desktop of the Solus Operating System.

### CODE REVIEW

**Comment by mar1n3r0** [comment]
Last chance guys, previous attempt I waited for 3 months before eventually deleting the package. I hope you will bring back the faith in contributing to my favorite distribution and open source project. 

In response, I will do my best to become a PR, respond in a timely manner and merge pull requests within foreseeable future.

**Comment by Duncaen** [comment]
Is there a way to not use git and the submodule, that is really counterproductive in case of caching for rebuilds?

**Comment by ebfe** [comment]
Remove the .xbps files and squash everything to 1 commit.
Please try to use the release tarballs, they contain the only submodule (gvc).
It's prefered to not use variables in `homepage`

**Comment by pullmoll** [comment]
There is a release v10.2.7 at https://github.com/solus-project/budgie-desktop/archive/v10.2.7.tar.gz so there's really no need to git clone, or is there?

**Comment by mar1n3r0** [comment]
Under no circumstance should you use GitHub generated tarballs to integrate Budgie Desktop into your project/distro/whatever-it-is. These tarballs are not correct, and are not supported by the maintainer.

If we determine your package uses GitHub generated tarballs, or bypasses our own release and build processes, your bug will be invalidated and no support will be received.

Note that a valid generated tarball for use from git can be generated and hosted in your own infrastructure as follows:

./autogen.sh
make distcheck

@ebfe What do you mean by remove .xbps files ? 

Also I tried squashing my last commits but they doubled. I used this command:

git reset --soft HEAD~4 &&
git commit

**Comment by ebfe** [comment]
There are non automatically generated release tarballs at https://github.com/solus-project/budgie-desktop/releases/download/v10.2.7/budgie-desktop-10.2.7.tar.xz.

**Comment by ebfe** [comment]
> @ebfe What do you mean by remove .xbps files ? 
the first commit (947e3c3) adds .xbps files in hostdir/binpkgs

**Comment by mar1n3r0** [comment]
Please see previous discussion on this:

https://github.com/voidlinux/void-packages/pull/4398

This template makes use of the tarballs which are specifically disa-vowed of by the project authors. This should be refactored to do a git checkout so that you can ensure you actually have all of the tarball since github does not fold in the submodules.

There should be no issues with Void policy on this matter since it is still a tagged, repeatable release that is made by the upstream project, only the transport mechanism is changing.

**Comment by ebfe** [comment]
There is a difference between autogenerated github tarballs and release tarballs that are uploaded by the maintainers...

**Comment by mar1n3r0** [comment]
So is this final or just a contradiction between reviewers ? Maybe discuss with @the-maldridge and then I can refactor as per agreed decision ?

**Comment by ebfe** [comment]
I don't know which tarballs you used in the initial PR and the commit is gone so i can't check. Maybe you used the autogenerated one and the real release tarball was overlooked.

**Comment by ebfe** [comment]
git rebase -i origin/master

assuming origin is the remote you use for https://github.com/voidlinux/void-packages
drop the commits that add .xbps files, squash the rest.

**Comment by mar1n3r0** [comment]
[DEBUG] Added `libtls10-2.3.6_1' into the dependency list (/host/binpkgs)
libcrypto37-2.3.6_1 (update) breaks installed pkg `libressl-2.3.8_1'
libssl38-2.3.6_1 (update) breaks installed pkg `libressl-2.3.8_1'
Transaction aborted due to unresolved dependencies.

I guess the bot is currently building some packages.

**Comment by ebfe** [comment]
> I guess the bot is currently building some packages.

No, you probably need to run `./xbps-src update-bootstrap` to update your masterdir

**Comment by mar1n3r0** [comment]
Hmm I ran it and it updated but it seems like the required ones are behind the installed ones, now that I think of it.

**Comment by mar1n3r0** [comment]
This is solved now thanks to @Duncaen. Basically I had to pull latest changes in masterdir first.
Here is what happens next:

ERROR: [trans] failed to download wget-1.18_1' package fromhttps://repo.voidlinux.eu/current/aarch64': Not Found
[DEBUG] [trans] failed to download binpkgs: Device or resource busy

**Comment by the-maldridge** [comment]
As long as the build is repeatable and is compliant with policy I don't care how the files are retrieved.

File: srcpkgs/budgie-desktop/template [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+mutter-devel gnome-menus-devel libwnck-devel
+libupower-glib1 vala libuuid-devel polkit-devel
+gettext-devel"
+depends="gnome-session"
+short_desc="Budgie desktop"
+maintainer="Frankie Wilde <train_me_well@abv.bg>"
+license="GPL-2, LGPL-2.1"
+homepage="https://github.com/solus-project/budgie-desktop"
+do_fetch() {
+	wget --no-check-certificate https://github.com/solus-project/budgie-desktop/releases/download/v${version}/${package}-${version}.tar.xz
```
Comment (ebfe) [inline srcpkgs/budgie-desktop/template]: no need to do this manually, just add it to distfiles.

File: srcpkgs/budgie-desktop/template [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	cd "${wrksrc}"
+	tar -xjvf "${pkgname}-${version}.tar.xz"
+}
+pre_configure() {
+	./autogen.sh
```
Comment (ebfe) [inline srcpkgs/budgie-desktop/template]: probably can be dropped. the release tarball already contains a configure script.

**Comment by mar1n3r0** [comment]
If I only knew how to do squashing properly. In this pull request I have 12 commits but when I choose git rebase -i HEAD~12 I see all commits on the upstream repository rather than only mine. After various attempts I am now brought to cherry-picking...

**Comment by ebfe** [comment]
```
git fetch origin
git rebase -i origin/master
```
assuming origin is the remote you use for https://github.com/voidlinux/void-packages
drop the commits that add .xbps files, squash the rest.

**Comment by pullmoll** [comment]
FWIW here is a [template that works](http://sprunge.us/HHhb) and should be correct.
I could just push it and make it close this PR?

**Comment by mar1n3r0** [comment]
Yes, please do. 

I feel a bit lost after diverting from my basic workflow with git.

**Comment by ebfe** [comment]
@mar1n3r0 HEAD~12 is probably not working like you think it does. Take a look at `man gitrevisions`.  @pullmoll Thanks :)

**Comment by mar1n3r0** [comment]
Even git rebase -i origin/master doesn't work as I think. When I execute it basically I see hundreds of commits which are not mine.

**Comment by mar1n3r0** [comment]
Thanks guys for your help and patience, this is my first package and still learning git.

**Comment by pullmoll** [comment]
FWIW here's my scheme to prepare and handle a git PR.

Do this once in your fork.
```
git remote add upstream git://github.com/voidlinux/void-packages.git
```
Now for a new package
```
git pull --rebase upstream master
git checkout -b mystuff
{ work on it and add files }
git add srcpkgs/mystuff
git commit
git push
{ more changes }
git add srcpkgs/mystuff
git commit --amend
git push -f
{ rebase to upstream master again, if there were changes like common/shlibs }
git pull --rebase upstream master
git push -f
```

**Comment by mar1n3r0** [comment]
Thanks that goes straight to my how-to articles. 

I was doing the fetch upstream rather than pull, maybe this is why it was outdated.

**Comment by Vaelatern** [comment]
Fetch downloads the upstream, but doesn't combine it with any branch. You need to manually merge or rebase for that to happen. Pull simply does a fetch and a merge all in one.

**Pull request closed without merging**